common/data/src/main/java/org/thingsboard/server/common/data/page/PageLink.java

```diff
@@ -68,16 +69,14 @@ public PageLink nextPageLink() {
     public Sort toSort(SortOrder sortOrder, Map<String,String> columnMap) {
         if (sortOrder == null) {
             return DEFAULT_SORT;
-        } else {
-            String property = sortOrder.getProperty();
-            if (columnMap.containsKey(property)) {
-                property = columnMap.get(property);
-            }
-            return Sort.by(Sort.Direction.fromString(sortOrder.getDirection().name()), property);
         }
+        return toSort(new ArrayList<>(List.of(sortOrder)), columnMap);
     }
 
     public Sort toSort(List<SortOrder> sortOrders, Map<String,String> columnMap) {
+        if (!isDefaultSortOrderAvailable(sortOrders)) {
+            sortOrders.add(new SortOrder(DEFAULT_SORT_PROPERTY, SortOrder.Direction.ASC));
```

Comment:
`sortOrders` list might be unmodifiable:
![image](https://github.com/thingsboard/thingsboard/assets/56742475/ac889666-c623-48ed-bce8-4e3aa8a2a1b4)

```
if (!isDefaultSortOrderAvailable(sortOrders)) {
  sortOrders = new ArrayList<>(sortOrders);
  sortOrders.add(...);
}
```